apps/ledger-live-desktop/src/renderer/families/internet_computer/StakeBanner.tsx

```diff
@@ -0,0 +1,132 @@
+import useFeature from "@ledgerhq/live-common/featureFlags/useFeature";
+import { AccountBanner } from "~/renderer/screens/account/AccountBanner";
+import React, { useCallback, useState } from "react";
+import { ICPAccount } from "@ledgerhq/live-common/families/internet_computer/types";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
+import { getAccountBridge } from "@ledgerhq/live-common/bridge/index";
+import { getBannerState } from "@ledgerhq/live-common/families/internet_computer/utils";
+
+const StakeBanner: React.FC<{ account: ICPAccount }> = ({ account }) => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const stakeAccountBanner = useFeature("stakeAccountBanner");
+  const [bannerState, setBannerState] = useState(getBannerState(account));
+
+  const updateBanner = useCallback(() => {
+    setBannerState(getBannerState(account));
+  }, [account]);
+
+  const onClickManageNeurons = useCallback(
+    (refresh = false) => {
+      if (account.type !== "Account") return;
+      dispatch(
+        openModal("MODAL_ICP_LIST_NEURONS", {
+          account,
+          refresh,
+        }),
+      );
+      updateBanner();
+    },
+    [account, dispatch, updateBanner],
+  );
+
+  const onClickStakeIcp = useCallback(() => {
+    const bridge = getAccountBridge(account, undefined);
+    const initTx = bridge.createTransaction(account);
+    dispatch(
+      openModal("MODAL_SEND", {
+        stepId: "amount",
+        account,
+        onConfirmationHandler: () =>
+          dispatch(
```

Comment:
so much chaining makes harder to read / refactor code in future, please split it

Reply:
fixed!